splunk/packages/edit-table/src/data.js

```diff
@@ -76,4 +77,43 @@ async function batchInsertKVEntries(collection, data, defaultErrorMsg, splunkApp
     return responseData;
 }
 
-export { updateKVEntry, getAllKVEntries, deleteAllKVEntries, batchInsertKVEntries };
+const executeJob = async (job, errorMessage) => {
+    try {
+        const response = await job.getResults().toPromise();
+        const errorMessages = response?.messages?.filter((msg) => msg.type === 'ERROR');
+        if (errorMessages && errorMessages.length > 0) {
+            console.error('Execute Job Error From Response', response);
+            throw new Error(errorMessages.map((msg) => msg.text).join('\n'));
+        }
+    } catch (err) {
+        console.error('Execute Job Error', err);
+        throw new Error(err?.message || errorMessage);
+    }
+};
+
+const createBackupForKvStore = async (splunkApp, kvStore, errorMessage) => {
+    const backupJob = SearchJob.create(
+        { search: `|inputlookup ${kvStore} |outputlookup ${kvStore}.bak.csv` },
+        { app: splunkApp }
+    );
+    return executeJob(backupJob, errorMessage);
+};
+
+const restoreKvStoreFromBackup = async (splunkApp, kvStore, errorMessage) => {
```

Comment:
it's a good idea that `splunkApp` is the first parameter here, but it should be the same for all the existing methods (e.g. `batchInsertKVEntries`) - The default value (`config.app`) can probably be removed - as far as I can see we always pass it now.